### ISSUE TITLE
[BUG] Water being deleted from left to right occurs instantly

### ISSUE THREAD

**Issue by Fluzzlesnuff** [Fluzzlesnuff/MinePlusPlus#14]
**Describe the bug**
Water is deleted from the left to the right instantly, as opposed to taking 5 ticks per block.

**Additional context**
Caused by updates running from left to right.

**Comment by Fluzzlesnuff** [comment]
# Working on a Fix
1. A block will be *scheduled* for deletion by changing it to its corresponding `U_DEL_WATER(number)` form. 
1. If a block is found to still be valid in any of the update steps, it will be turned back into its normal form. 
1. When checking whether a block is valid, adjacent `U_DEL_WATER` will be treated as normal water. 
1. At the very end of water deletion updating, blocks will actually be deleted.
---
This system will prevent a chain reaction of deletions occurring in each update, and is similar in purpose and implementation to the `GEN_T` system.

**Comment by Fluzzlesnuff** [comment]
`U_DEL_WATER` blocks will be added after the ID reorganization process is completed. Only then will I attempt to fix this bug. For now, this bug is the smallest of my worries; the program doesn't even compile due to the ID reorganization.

**Comment by Fluzzlesnuff** [comment]
This bug was fixed much as I explained in my earlier comment. 
1. Blocks are scheduled for deletion, but use `Blocks::Update::deletedWater#` IDs.
2. Blocks are, for now, never changed back once scheduled for deletion.
3. `block.isWater()` returns `true` even if the specified block is scheduled for deletion.
4. Same as original plan.

This system causes water to flow backwards towards its original source block when the source block is destroyed (it's a bit hard to explain), much like in Minecraft.
